src/test/java/org/killbill/billing/plugin/vertex/VertexTaxCalculatorTest.java

```diff
@@ -231,6 +231,40 @@ public void testTaxDescription() throws Exception {
         assertEquals("CA STATE TAX", result.get(0).getDescription());
     }
 
+    @Test(groups = "fast")
+    public void testTaxEffectiveRate() throws Exception {
+        //given
+        given(invoice.getInvoiceItems()).willReturn(Collections.singletonList(taxableInvoiceItem));
```

Comment:
can we break into 3 test cases - , no tax rate returned, 0 tax rate returned , non-zero tax rate returned

Reply:
I added few test cases and the one showing the behaviour when the amount is 0 (usually this is what happens when taxRate is 0). This should be regular scenario - https://github.com/killbill/killbill-vertex-plugin/blob/a1a9e5f6ece10d3aa332a3ffbc2bbc1831da8357/src/test/java/org/killbill/billing/plugin/vertex/VertexTaxCalculatorTest.java#L280, but this one is an exception - https://github.com/killbill/killbill-vertex-plugin/blob/a1a9e5f6ece10d3aa332a3ffbc2bbc1831da8357/src/test/java/org/killbill/billing/plugin/vertex/VertexTaxCalculatorTest.java#L297 and

Reply:
done